### PR TITLE
Lets xenos turn off rail flashlights by slashing the gun

### LINKED ISSUE #7438: Rail flashlights are too easy to use compared to flares
### Description

As it turns out, flares have some hefty competition, in the form of rail flashlights. 
As far as i can tell, pocket pistols with a flashlight are easier to hide than flares, smaller than them, smaller than normal flashlights, produces as much light and they do not ever run out. 

I've been seeing it for a couple games now, marines just throw guns with lights attached to them, and it really seems to work well for them. 

image

### Test Merges

Server-Side Code Modifications: https://tgstation13.org/parsed-logs/sybil/server_side_modifications.dm

Round ID: [9855](https://scrubby.melonmesa.com/round/9855)
BYOND Version: 514.1557
Server revision compiled on: 
Master commit: 72bfd5ea7400cea5463a6f6a1596f4fbcddda411
The following pull requests are currently test merged:
#7383: 'Fixes destroyed limbs also being broken' by QualityVan at commit debc9b201a
#7373: 'Fixes Corrosive Acid Melt Times and Process Time Adjustment Math' by Surrealaser at commit 2f26ef06be
#7099: 'Hemodile Buff' by Surrealaser at commit 59be3ff081
#7194: 'Prevent too many xeno at start' by BraveMole at commit a09e59266a
#7257: 'Clawed Huggers no longer suck' by Surrealaser at commit 3a7df2226b
#7181: 'Losing armor grants speed' by Wayland-Smithy at commit 64b0b0b498
#7351: 'adds an iff-less heavy machinegun' by Pariah919 at commit db076f935f
#7377: 'Removes double healing from most xeno sources' by QualityVan at commit 84ea472f97
#7372: 'Reverts old MJP sentries change that literally killed sentries, with less sunder.' by SplinterGP at commit 00201d901e

Local commit: 64ca85def1da5dbb1c704b6f1a22ad1a479b400c
Server tools version: /tg/station 13 Server v3.2.6.2

### Reproduction Steps

Play.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by QualityVan**
## About The Pull Request
Xenos can now attack floor guns to turn off the rail light if there is one.

## Why It's Good For The Game
Tired of infinite pocket pistols lying around providing light forever until one of the castes with acid clears it. If you want lights to leave around as a marine bring flares.
Closes #7438 

## Changelog
:cl:
add: Slashing guns as a xeno will turn any lights on them off
/:cl:

### CODE REVIEW

**Comment by vilereaver41** [comment]
<3 thx man

**Comment by lbnesquik** [comment]
Thank you!

**Comment by RuneQuester** [comment]
Hello? Based Department?

**Comment by TheRealMedalis** [comment]
Wouldn't it be easier to just remove the rail flashlight attachment from the pocket pistol options?

**Comment by David-Stormwell** [comment]
> 
> 
> Wouldn't it be easier to just remove the rail flashlight attachment from the pocket pistol options?

Yes, but that would've not had solved the issue, marine mains would in fact just use other guns.

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by joniroq** [comment]
Would it not be simpler to have the raillight turn off when the weapon is dropped? Armor lights turn off then removed so why not gun lights

File: code/modules/projectiles/gun_system.dm [hunk 1]
```diff
@@ -1177,3 +1177,11 @@ and you're good to go.
 		return
 	active_attachable.fire_attachment(target, src, gun_user) //Fire it.
 	last_fired = world.time
+
+///For letting xenos turn off the flashlights on any guns left lying around.
```
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/modules/projectiles/gun_system.dm:1181]:
```suggestion
//For letting xenos turn off the flashlights on any guns left lying around.
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**